### PR TITLE
[wifiled] Fix ArrayIndexOutOfBoundsException during discovery

### LINKED ISSUE #6848: [wifiled] Scheduled runnable ended with exception
## Current Behavior
It does not seem to affect functionality, but when OH starts, there is an exception...

```
2020-01-17 07:11:09.780 [WARN ] [org.eclipse.smarthome.core.internal.common.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
java.lang.ArrayIndexOutOfBoundsException: 1
        at org.openhab.binding.wifiled.internal.discovery.WiFiLEDDiscoveryService.discover(WiFiLEDDiscoveryService.java:107) ~[?:?]
        at org.openhab.binding.wifiled.internal.discovery.WiFiLEDDiscoveryService.lambda$0(WiFiLEDDiscoveryService.java:64) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_232]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_232]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_232]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[?:1.8.0_232]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232
```

## Steps to Reproduce (for Bugs)
I have 2 Arilux AL-LC02 controllers configured as LD382A Things. All I do is restart OH to reproduce the issue. This has been happening for a long while.

## Your Environment

* Version used: (e.g., openHAB and add-on versions) OH 2.5.1-2
* Operating System and version (desktop or mobile, Windows 10, Raspbian Buster, ...): FedoraServer 31

### PR DESCRIPTION
**Pull request by 9037568**
Fix the ArrayIndexOutOfBoundsException.

This change also:
     allows the discovery process to skip over unparseable responses and continue searching
     adds additional logging output
     fixes a few tiny typos
     fixes a few tiny problems in the README

A [test jar](https://drive.google.com/open?id=1WXviT-zKdU8kSeqXn630UpF0jgVTbZTT) is available.

Fixes #6848 

Signed-off-by: 9037568 <namraccr@gmail.com>

### CODE REVIEW

**Comment by openhab-5iver** [comment]
Thank you... I'll test tonight.

File: bundles/org.openhab.binding.wifiled/README.md [hunk 1]
```diff
@@ -42,24 +42,24 @@ You can choose between two drivers with different functionality:
...
+without any fading effect.
 
-If you prefer to switch colors smoothly and to turn your light on and off by slightly rising/decreasing the brightness
-you should try the FADING driver.
-If selected you can also set the number of fading steps and the fading duration in the thing configuration.
-Note that each fading step will at least take 10 ms for being processed.
-This natural limit is given by the speed of the LED controller and your network speed.
-Thus, a color fading with a configured fading duration of 0s might still take some time (count with more than 1 second for 100 steps).
-IF the "FADING" driver is chosen the program channel and the programSpeed channel will not have any effect.
+If you prefer to switch colors smoothly and to turn your light on and off by slightly increasing/decreasing the brightness,
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.wifiled/README.md]: Please put each sentence on a single line.

File: bundles/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/internal/discovery/WiFiLEDDiscoveryService.java [hunk 2]
```diff
@@ -92,17 +92,23 @@ private synchronized void discover() {
...
                 byte[] data = packet.getData();
                 String s = bytesToString(data);
-                logger.debug("Disover response received: '{}' [{}] ", s, ClassicWiFiLEDDriver.bytesToHex(data));
+                logger.debug("Discovery response received: '{}' [{}] ", s, ClassicWiFiLEDDriver.bytesToHex(data));
 
                 // 192.168.178.25,ACCF23489C9A,HF-LPB100-ZJ200
                 // ^-IP..........,^-MAC.......,^-HOSTNAME.....
 
                 String[] ss = s.split(",");
+                if( ss.length < 3 ) {
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/internal/discovery/WiFiLEDDiscoveryService.java]:
```suggestion
                if (ss.length < 3 ) {
```

File: bundles/org.openhab.binding.wifiled/README.md (hunk 1 above)
Comment (9037568) [reply bundles/org.openhab.binding.wifiled/README.md]: Why?

File: bundles/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/internal/discovery/WiFiLEDDiscoveryService.java (hunk 2 above)
Comment (9037568) [reply bundles/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/internal/discovery/WiFiLEDDiscoveryService.java]: Done.

File: bundles/org.openhab.binding.wifiled/README.md (hunk 1 above)
Comment (J-N-K) [reply bundles/org.openhab.binding.wifiled/README.md]: That makes it easier for review later on.

Comment (9037568) [reply bundles/org.openhab.binding.wifiled/README.md]: We shouldn't write a README with one sentence per line any more than we should write a book with one sentence per line.

Comment (J-N-K) [reply bundles/org.openhab.binding.wifiled/README.md]: The lines are auto-concatenated for displaying, so the user sees them as one paragraph even if they are separate lines in the file (like in HTML files).

Comment (9037568) [reply bundles/org.openhab.binding.wifiled/README.md]: I've changed this back only because it was that way originally.

**Comment by Hilbrand** [comment]
@openhab-5iver Did you had time to test this?

**Comment by openhab-5iver** [comment]
Sorry... I forgot to come back! Yes, I've built a jar with these changes and have not seen the error on OH restart or discovery. Thank you, Chris!

**Comment by hpd96** [comment]
PASSED
thanks

**Pull request merged**